mempool/mempool_test.go

```diff
@@ -795,6 +795,67 @@ func TestDoubleSpends(t *testing.T) {
 	testPoolMembership(tc, tx2, false, true, false)
 }
 
+func TestDoubleSpendsFromDAG(t *testing.T) {
+	tc, spendableOuts, teardownFunc, err := newPoolHarness(t, &dagconfig.SimnetParams, 2, "TestDoubleSpendsFromDAG")
+	if err != nil {
+		t.Fatalf("unable to create test pool: %v", err)
+	}
+	defer teardownFunc()
+	harness := tc.harness
+
+	tx, err := harness.createTx(spendableOuts[0], uint64(txRelayFeeForTest), 1)
+	if err != nil {
+		t.Fatalf("unable to create transaction: %v", err)
+	}
+
+	dag := harness.txPool.cfg.DAG
+	block, err := mining.PrepareBlockForTest(dag, harness.dagParams, dag.TipHashes(), []*wire.MsgTx{tx.MsgTx()}, false)
+	if err != nil {
+		t.Fatalf("PrepareBlockForTest: %v", err)
+	}
+	isOrphan, isDelayed, err := dag.ProcessBlock(util.NewBlock(block), blockdag.BFNoPoWCheck)
+	if err != nil {
+		t.Fatalf("ProcessBlock: %v", err)
+	}
+	if isDelayed {
+		t.Fatalf("ProcessBlock: block " +
+			"is too far in the future")
+	}
+	if isOrphan {
+		t.Fatalf("ProcessBlock: block got unexpectedly orphaned")
+	}
+
+	// Check that a transaction that some of its outputs exists in the DAG UTXO
```

Comment:
I'm not sure I understand what "some of its outputs exists in the DAG UTXO" means